dags/utils/backoff.py

```diff
@@ -0,0 +1,24 @@
+import time
+import logging
+import random
+import sys
+
+logger = logging.getLogger()
+
+def retry_with_backoff(fn, retries = 10, backoff_in_seconds = 1):
+    try:
+        return fn()
+    except Exception as e:
+        print()
+        if retries <= 0:
+            raise
+
+        sleep = (backoff_in_seconds * 2 ** retries +
+                random.uniform(0, 1))
```

Comment:
The retries now happen in reverse order, since we count down to zero. We can solve that by having 2 parameters in the function (for example both a retry counter and a max_retries value, or a retries_left and a max_retries)